### PR TITLE
[Merged by Bors] - feat: Generalize results of `CanonicalEmbedding` to fractional ideals

### PR DESCRIPTION
**Pull request by xroblot**
The main results of `NumberTheory.NumberField.CanonicalEmbedding`, that is `exists_ne_zero_mem_ringOfIntegers_lt` and `exists_ne_zero_mem_ringOfIntegers_of_norm_le` about the existence of algebraic integers satisfying special properties, are generalized from the ring of integers to fractional ideals. 

---

- [x] depends on: #9836 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/NumberTheory/NumberField/Discriminant.lean [hunk 1]
```diff
@@ -17,7 +17,7 @@ This file defines the discriminant of a number field.
 
 ## Main result
 
-* `NumberField.discr_gt_one`: **Hermite-Minkowski Theorem**. A nontrivial number field has
+* `NumberField.abs_discr_gt_two`: **Hermite-Minkowski Theorem**. A nontrivial number field has
 nontrivial discriminant.
```
Comment (riccardobrasca) [inline Mathlib/NumberTheory/NumberField/Discriminant.lean:21]:
```suggestion
* `NumberField.abs_discr_gt_two`: **Hermite-Minkowski Theorem**. A nontrivial number field has
discriminant greater than `2`.
```
Same in the theorem below.

File: Mathlib/NumberTheory/NumberField/CanonicalEmbedding.lean [hunk 2]
```diff
@@ -428,6 +431,80 @@ theorem mem_span_latticeBasis [NumberField K] (x : (E K)) :
...
+  have : f (integralBasis K i) = ((latticeBasis K).restrictScalars ℚ) i := by
+    apply Subtype.val_injective
+    rw [LinearMap.codRestrict_apply, AlgHom.toLinearMap_apply, Basis.restrictScalars_apply,
+      latticeBasis_apply]
+    rfl
+  simp_rw [LinearMap.coe_comp, LinearEquiv.coe_coe, Function.comp_apply, this, Basis.repr_self]
+
+variable (I : (FractionalIdeal (𝓞 K)⁰ K)ˣ)
+
+theorem det_basisOfFractionalIdeal_eq_norm
```
Comment (riccardobrasca) [inline Mathlib/NumberTheory/NumberField/CanonicalEmbedding.lean:460]: Can you mention [NumberField.det_basisOfFractionalIdeal_eq_absNorm](https://leanprover-community.github.io/mathlib4_docs/Mathlib/NumberTheory/NumberField/FractionalIdeal.html#NumberField.det_basisOfFractionalIdeal_eq_absNorm) here? I mean, explaining what is the mathematical difference. Maybe even a docstring at the beginning of the section is a good idea.

**Review by riccardobrasca — APPROVED** [review]
Thanks!

bors d+

File: Mathlib/NumberTheory/NumberField/CanonicalEmbedding.lean (hunk 2 above)
Comment (xroblot) [reply Mathlib/NumberTheory/NumberField/CanonicalEmbedding.lean:466]: I am not sure what you have in mind. I put a docstring explaining why the result means (for me) and why it is useful. Please let me know if you're happy with that before I merge this PR. 

Thanks.

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**